src/common/telemetry.ts

```diff
@@ -146,35 +144,15 @@ export class BaseTelemetryReporter extends Disposable {
 	private userOptIn = false;
 	private errorOptIn = false;
 	private _extension: vscode.Extension<any> | undefined;
+	private readonly appenders = new Map<string, ITelemetryAppender>();
```

Comment:
I saw a thread in slack about supporting multiple host, but that should be discussed and addressed in a separate PR for the whole extension, not mix changes here

Reply:
it is not about multi host, it is about analytics to corresponding host

Reply:
You can just use `getGitpodHost` which reads it from setting, no need to pass it around as the first thing we do  is update the settings when trying to connect to a new workspace before sending any telemetry

> it is reporting about analytics to corresponding host, never around or to another.

We only support one host currently, the first action when connecting to a new workspace is [sign in](https://github.com/gitpod-io/gitpod-vscode-desktop/blob/ff65e7992ff40472ac4c7e24a4be05db1f664aad/src/remoteConnector.ts#L641) which internally updates `gitpodHost` setting, also there's a [notification when user has a remote workspace already connected](https://github.com/gitpod-io/gitpod-vscode-desktop/blob/ff65e7992ff40472ac4c7e24a4be05db1f664aad/src/remoteSession.ts#L44-L50) which can be improved

Reply:
> You can just use getGitpodHost

I don't think it is true. For instance in https://github.com/gitpod-io/gitpod-vscode-desktop/blob/ff65e7992ff40472ac4c7e24a4be05db1f664aad/src/authentication/authentication.ts#L218 gitpod service should be fixed during all execution, but since code is async if for some reason config is changed during event ticks telemetry will leak to another host, we should make sure that never happens.